jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/ResourceCollection.java

```diff
@@ -257,6 +257,13 @@ public boolean isDirectory()
         return true;
     }
 
+    @Override
+    public boolean isReadable()
+    {
+        // always a directory, never readable
+        return false;
+    }
```

Comment:
A directory can be readable, in that it's contents can be listed.

I can see an interpretation of `isReadable` that means there is content available, but this is a change in semantics that I don't think we need.    It is a valid check to ask a directory if it is readable or not.   Perhaps this check could be made in the constructor?

Reply:
Asking "isReadable()" should be different than "isDirectory()".
Meaning if you are `isDirectory() == true` then you have no need to call `isReadable()` anyway.

If you look at usage, the `isReadable()` checks are typically for places where we expect a non-directory that can be accessed and be read (eg: a file)

Reply:
I know that we used to check some directories for readability.   I don't really see the need for changing this semantic.   What does a directory that exists but is not actually readable on the file system look like?  If all our directory handling code expects directories to not be readable, then we won't be able to distinguish this case?

Reply:
perhaps rename to `hasContent` or `hasReadableContent` if you want your semantic, to distinguish from the `Files` method that can apply to a directory.

Reply:
Perhaps it should be `canBeOpenForRead()` instead.
As it's not meant for "can we access" or "do we have read permissions".